### PR TITLE
Implement data slice method

### PR DESCRIPTION
**Pull request by jgrewe**
... upon successful build and merge we may want to pick these changes to 1.4 as well.

### CODE REVIEW

**Comment by jgrewe** [comment]
@gicmo and @achilleas-k: no more warnings, ready to merge?

**Comment by achilleas-k** [comment]
I'm good to merge is @gicmo is happy with the changes he requested.

File: src/util/dataAccess.cpp [hunk 1]
```diff
@@ -324,6 +324,43 @@ bool positionAndExtentInData(const DataArray &data, const NDSize &position, cons
...
+DataView dataSlice(const DataArray &array, const std::vector<double> &start, const std::vector<double> &end,
+                   const std::vector<std::string> &units) {
+    if (array == nix::none) {
+        throw UninitializedEntity();
+    }
+    if (start.size() == 0 || start.size() != end.size()) {
+        throw OutOfBounds("Number of start entries does not match number of end entries.", 0);
+    }
+    if (start.size() != array.dimensionCount()) {
+        throw OutOfBounds("Number of start/end entries does not match dimensionality of the data.", 0);
```
Comment (gicmo) [inline src/util/dataAccess.cpp]: We could pass `start.size()` as the "where" parameter (currently `0`). But maybe that is confusing.

Comment (gicmo) [reply src/util/dataAccess.cpp]: Actually, I think `OutOfBounds` is probably not 100% the right exception, but I don't think that is crucial.

**Comment by gicmo** [comment]
Question is, do we bother finding better exceptions for the checks or not?

**Comment by achilleas-k** [comment]
InvalidRank?

File: src/util/dataAccess.cpp [hunk 2]
```diff
@@ -324,6 +324,43 @@ bool positionAndExtentInData(const DataArray &data, const NDSize &position, cons
...
+    }
+    if (units.size() > 0 && units.size() != start.size()) {
+        throw OutOfBounds("Number of units does not match dimensionality of the data.", 0);
```
Comment (gicmo) [inline src/util/dataAccess.cpp]: Technically also not `OutOfBounds`, I think.

**Pull request merged**